src/Shared/Traits.cs

```diff
@@ -32,6 +32,15 @@ internal class EscapeHatches
 
         public readonly bool DoNotLogProjectImports = Environment.GetEnvironmentVariable("MSBUILDDONOTLOGIMPORTS") == "1";
 
+        public readonly bool CacheAssemblyInformation =
+#if FEATURE_ASSEMBLY_LOADFROM
+            // On full framework, default to old/well-tested/known-good behavior, but allow an opt in.
+            Environment.GetEnvironmentVariable("MSBUILDCACHEASSEMBLYINFORMATION") == "1";
+#else
+            // On .NET Core, go for speed, but allow an opt out.
+            Environment.GetEnvironmentVariable("MSBUILDDONOTCACHEASSEMBLYINFORMATION") == "1";
```

Comment:
Shouldn't this be different instead of equals? Cache when ("Do not cache" is not 1)